hie-plugin-api/Haskell/Ide/Engine/PluginsIdeMonads.hs

```diff
@@ -309,16 +308,18 @@ toDynJSON = CD.toDyn
 runPluginCommand :: PluginId -> CommandId -> Value
                   -> IdeGhcM (IdeResult DynamicJSON)
 runPluginCommand p com arg = do
+  let PluginId p' = p
+      CommandId com' = com
```

Comment:
Thanks, but a minor nit. For `mkLspCmdId` just above you match the `PluginId` and `CommandId` directly as parameters.  Why is this different? if you need the original, perhaps do it as

```haskell
runPluginCommand p@(PluginId p') com@(CommandId cid) arg = do
```